tests/test_triples_factory.py

```diff
@@ -397,3 +398,48 @@ def test_inverse_triples(self):
         )
 
         self.assertIn(f'likes{INVERSE_SUFFIX}', triples_factory.relation_to_id)
+
+
+def test_get_absolute_split_sizes():
+    """Test get_absolute_split_sizes."""
+    for num_splits, n_total in zip(
+        (2, 3, 4),
+        (100, 200, 10412),
+    ):
+        # generate random ratios
+        ratios = np.random.uniform(size=(num_splits,))
+        ratios = ratios / ratios.sum()
+        sizes = get_absolute_split_sizes(n_total=n_total, ratios=ratios)
+        # check size
+        assert len(sizes) == len(ratios)
+
+        # check value range
+        assert all(0 <= size <= n_total for size in sizes)
+
+        # check total split
+        assert sum(sizes) == n_total
+
+        # check consistency with ratios
+        rel_size = np.asarray(sizes) / n_total
+        decimal = np.floor(-np.log10(1 / n_total))
```

Comment:
are you doing `-np.log10(1 / n_total)` to check for numeric stability? otherwise you could directly do `np.log10(n_total)`

Reply:
96d745b  and 371add8